### PR TITLE
NRP-2904 - Added new endpoint for research projects

### PR DESCRIPTION
**Pull request by lgrzybowski**
I added new endpoint for research project. 
__Results__ for:

 _rubocop:_
`147 files inspected, no offenses detected`

_rspec:_
`Finished in 0.06084 seconds (files took 2.53 seconds to load)
140 examples, 0 failures`

### CODE REVIEW

**Comment by lgrzybowski** [comment]
@jonathanchrisp It looks like ruby head does not have ruby-head as an allowed failure in travis yml. Which can make a builds fail sometimes.  

Build works fine on my local machine with ruby version `ruby 2.1.3p242 (2014-09-19 revision 47630) [x86_64-darwin13.0]`
 
@benSlaughter Advices to use configuration like:

```
rvm:
- ruby-head
- 2.1.0
- 2.0.0
- 1.9.3
- 1.9.2
matrix:
  allow_failures:
  - rvm: ruby-head
```

File: lib/bwapi/client/projects.rb [hunk 1]
```diff
@@ -7,6 +7,7 @@
 require 'bwapi/client/projects/facebook_queries'
 require 'bwapi/client/projects/group'
 require 'bwapi/client/projects/preview_search'
+require 'bwapi/client/projects/research_project.rb'
```
Comment (jonathanchrisp) [inline lib/bwapi/client/projects.rb]: we don't need the `.rb` here

Comment (jonathanchrisp) [inline lib/bwapi/client/projects.rb]: we should also just call this file `research` as the sub path is just `/research`

**Comment by jonathanchrisp** [comment]
@lgrzybowski / @benSlaughter - we [used](https://github.com/jonathanchrisp/bwapi/commit/be3d06857400be7584624cc32e4b6b8d28724aec) to allow ruby-head to fail as it was quite unstable although it did get a lot better hence me removing it. I'm happy to do add this back in :smile: 

If possible can you also add in `- 2.2.0` into the list of versions to test also.

**Comment by lgrzybowski** [comment]
@jonathanchrisp  Done.

**Comment by jonathanchrisp** [comment]
@lgrzybowski - please rebase into a single commit and then ask @benSlaughter to merge as he has the permissions.

**Comment by lgrzybowski** [comment]
We moved that PR to private repo.

**Pull request closed without merging**